sljit_src/sljitNativeMIPS_common.c

```diff
@@ -353,6 +366,60 @@ static sljit_s32 function_check_is_freg(struct sljit_compiler *compiler, sljit_s
 
 #endif /* SLJIT_CONFIG_MIPS_32 && SLJIT_ARGUMENT_CHECKS */
 
+static void get_cpu_features(void)
+{
+#if defined(__GNUC__)
+#if !defined(SLJIT_IS_FPU_AVAILABLE)
+	sljit_u32 fir;
+#endif
+#if defined(SLJIT_CONFIG_MIPS_32) && SLJIT_CONFIG_MIPS_32
+#if (!defined(SLJIT_DETECT_FR) || (defined(SLJIT_DETECT_FR) && SLJIT_DETECT_FR == 1)) \
+	&& (!defined(SLJIT_DETECT_FR) || SLJIT_DETECT_FR == 1) \
+	&& (!defined(SLJIT_MIPS_REV) || (defined(SLJIT_MIPS_REV) && SLJIT_MIPS_REV < 6))
+	sljit_sw flag = -1;
+#endif
+#endif /* SLJIT_CONFIG_MIPS_32 */
+#endif /* __GNUC__ */
+	int feature_list = CPU_FEATURE_DETECTED;
+
+#if defined(__GNUC__) && !defined(SLJIT_IS_FPU_AVAILABLE)
+	__asm__ ("cfc1 %0, $0" : "=r"(fir));
+#endif
+	if ((fir & (1 << 22)) != 0)
+		feature_list |= CPU_FEATURE_FPU;
+
+	if (!(feature_list & CPU_FEATURE_FPU) &&
+		(fir & (1 << 17)) != 0 && (fir & (1 << 21)) != 0)
+		feature_list |= CPU_FEATURE_FP32;
+
+#if defined(SLJIT_CONFIG_MIPS_32) && SLJIT_CONFIG_MIPS_32
+	if ((feature_list & CPU_FEATURE_FPU) && ((feature_list & CPU_FEATURE_FP32) == 0)) {
+#if defined(SLJIT_MIPS_REV) && SLJIT_MIPS_REV >= 6
+		feature_list |= CPU_FEATURE_FR;
+#elif defined(SLJIT_DETECT_FR) && SLJIT_DETECT_FR == 0
+#if defined(SLJIT_MIPS_REV) && SLJIT_MIPS_REV >= 5
+		feature_list |= CPU_FEATURE_FR;
```

Comment:
`SLJIT_DETECT_FR` is not defined by default, and therefore the fact that it is set to off is the most relevant part of this test.

if it was set to ON, then the detection will be done even when SLJIT_MIPS_REV == 5, but never when SLJIT_MIPS_REV >= 6 since it is no longer relevant per the documentation.

Note that the detection is basically a "hack".  The "correct" way will be to look for the StatusFR flag directly (bit 26 in the status register), but doing so requires a privileged instruction (except in MIPSr5, but in a way that is no longer supported by MIPSr6) and the "only" way that would be compatible up to MIPSr2 is to `ptrace(GETREGS)`.